omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java

```diff
@@ -124,11 +128,14 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
 				}
 				
-				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
+				if (user == null) {
+					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "auth.question.fill");
+					request.setAttribute("secretQuestion", getRandomFakeSecretQuestion(username));
```

Comment:
It was asked to implement it this way as it was mentioned in the ticket
comments. Is that alright?
On 04-Jan-2016 8:49 pm, "Wyclif Luyima" <notifications@github.com> wrote:

> In
> omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java
> <https://github.com/openmrs/openmrs-module-legacyui/pull/20#discussion_r48740132>
> :
>
> > @@ -124,11 +128,14 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
> >  					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
> >  				}
> >  				
> > -				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
> > +				if (user == null) {
> > +					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "auth.question.fill");
> > +					request.setAttribute("secretQuestion", getRandomFakeSecretQuestion(username));
>
> Why ask them a fake question? I would just keep prompting them for the
> username? Is this the general behavior on other sites?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-legacyui/pull/20/files#r48740132>
> .
>